src/sys/socket/mod.rs

```diff
@@ -227,8 +227,20 @@ pub enum SockProtocol {
     #[cfg(any(target_os = "android", target_os = "linux"))]
     #[cfg_attr(docsrs, doc(cfg(all())))]
     EthAll = libc::ETH_P_ALL.to_be(),
+    /// The Controller Area Network raw socket protocol
+    /// ([ref](https://docs.kernel.org/networking/can.html#how-to-use-socketcan))
+    #[cfg(target_os = "linux")]
+    #[cfg_attr(docsrs, doc(cfg(all())))]
+    CanRaw = libc::CAN_RAW,
 }
 
+/// The Controller Area Network broadcast manager protocol
+/// ([ref](https://docs.kernel.org/networking/can.html#how-to-use-socketcan))
+#[cfg(target_os = "linux")]
+#[cfg_attr(docsrs, doc(cfg(all())))]
+#[allow(non_upper_case_globals)]
+pub const CanBcm: SockProtocol = SockProtocol::NetlinkUserSock; // Matches libc::CAN_BCM
+
```

Comment:
It will be easier for users to import if you do it like this
```suggestion
impl SockProtocol {
    /// The Controller Area Network broadcast manager protocol
    /// ([ref](https://docs.kernel.org/networking/can.html#how-to-use-socketcan))
    #[cfg(target_os = "linux")]
    #[cfg_attr(docsrs, doc(cfg(all())))]
    #[allow(non_upper_case_globals)]
    pub const CanBcm: SockProtocol = SockProtocol::NetlinkUserSock; // Matches libc::CAN_BCM
}
```

Reply:
Oh, wow. It never even occurred to me that it could be done this way! Very nice.